test/e2e/storage/pvc_protection.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	storage "k8s.io/api/storage/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/util/slice"
+	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = SIGDescribe("PVC Protection [Feature:PVCProtection]", func() {
+	var (
+		client       clientset.Interface
+		nameSpace    string
+		err          error
+		pvc          *v1.PersistentVolumeClaim
+		storageClass *storage.StorageClass
+	)
+
+	f := framework.NewDefaultFramework("pvc-protection")
+	BeforeEach(func() {
+		skipUnlessPVCProtectionEnabled()
+		client = f.ClientSet
+		nameSpace = f.Namespace.Name
+		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(client, framework.TestContext.NodeSchedulableTimeout))
+
+		By("Creating a Storage Class")
+		testStorageClass := storageClassTest{
+			name:      "default",
+			claimSize: "1Gi",
+		}
+		suffix := "pvc-protection"
+		storageClass = newStorageClass(testStorageClass, nameSpace, suffix)
+		storageClass, err = client.StorageV1().StorageClasses().Create(storageClass)
+		Expect(err).NotTo(HaveOccurred(), "Error creating storage class")
```

Comment:
Any particular reason this test creates new storage class rather than using the one that already exists?

Reply:
I don't know that one already exist.
Is it a good idea to assume that a Storage Class exist?

Reply:
There should already be a default storageclass available in all environments. I think plenty of tests assume that, so we would be fine.